### ISSUE TITLE
Any way to send a message outside an event

### ISSUE THREAD

**Issue by rasikhq** [Roughsketch/libdiscord#4]
I want to send a message to the channel that is first in the channel list, however when I get the channel list in on_ready the .size() of the vector is 0 and there are no channels to send message to.

```
bot->on_ready([bot]() {
	std::vector<discord::channel> channels = bot->channels();
	std::cout << "Total channels: " << channels.size() << std::endl;
});
```
I tried delaying the channels() by adding Sleep but that doesn't seem to help (Tested if its due to a small delay or something)

**Comment by Roughsketch** [comment]
This library isn't complete, nor is it maintained so if you are using this for a real bot I'd recommend finding another library.

However, the answer is that this library does not handle all events right now. If it was more complete, there would be an event for handling new channels (so you could do something like `on_channel_join`) but as it is right now it just mainly handles messages.

**Comment by rasikhq** [comment]
This library is very straight forward, after trying and looking for many others this was the only one I was able to get up and running without weird errors, well I guess its a dead end then.

I only wanted to do simple message sending and receiving, nothing more complicated, however the only thing I'm missing right now is that I just want to send a message whenever the bot gets "alive", like "Hey I'm now active" when the program starts.

**Comment by Roughsketch** [comment]
Yeah, the C++ ecosystem is pretty poor for Discord I think. I'm glad you think this library is straight forward though.

If you still want to use it, you are free to fork and add the functionality you need. Its been a long time since I've looked at this project, but I think it shouldn't be too complex.

Alternatively, I could recommend a library in a different language which is personally my main library for all my Discord bots. It's another project I've contributed to called [serenity](https://github.com/serenity-rs/serenity). However it is Rust, not C++ which may be an issue if you're wanting to use C++ in particular.

**Comment by rasikhq** [comment]
Solved as in #5 
---
Edit: I can't believed I managed to do it :D, checked out some discord documentation on the various events, they mentioned that Ready won't have the channels/guilds ready, however GUILD_CREATED is also invoked when they are available, so I tried to add it into the source and it worked! Might fork and make a pull request

I'm bound to use C++, i'll try to add functionality to this one in a fork however my knowledge in 'advanced' stuff is quite low, more like inexperienced, this lib is perfect for me even in its current form, I just need to send a message when the bot gets alive which is the only thing I need in this right now.

Anyhow I'd appreciate if you could spare some time to just fix/update so one could send a message when bot is ready or something, until then I'll try my luck.

Thanks!

P.S: Where did you get the event ids to use, for example
